skimage/measure/fit.py

```diff
@@ -868,7 +868,7 @@ def ransac(data, model_class, min_samples, residual_threshold,
                 break
 
     # estimate final model using all inliers
-    if best_inliers is not None:
+    if best_inliers is not None and sum(best_inliers) > 0:
         # select inliers for each data array
         data_inliers = [d[best_inliers] for d in data]
         best_model.estimate(*data_inliers)
```

Comment:
Should we add an `else` clause to set `best_model=None` when there are no inliers to use in estimating the model? Or maybe raise a UserWarning in that case? (I am not a `ransac` user, so am not sure what the best solution is)

Reply:
I would also add an `else` clause. Morally (haven't checked for types):
```py
if any(best_inliers):
    # estimate model
else:
    best_inliers = None
    best_model = None
    # warn user
```
and update test to check for `None` return values.

Reply:
@grlee77 @mkcor please resolve the convo if the change is ok for you

Reply:
@emmanuelle I actually don't have 'resolve' permission -- is it possible? Also, I admit that I would love to see type hints in the `ransac` function... because an object could be `not None` but not iterable either...